sw/device/lib/crypto/impl/ecc/ecdh_p256.c

```diff
@@ -105,3 +107,47 @@ status_t ecdh_p256_shared_key_finalize(ecdh_p256_shared_key_t *shared_key) {
 
   return OTCRYPTO_OK;
 }
+
+status_t ecdh_p256_sideload_keypair_start(void) {
+  // Load the ECDSA/P-256 app. Fails if OTBN is non-idle.
+  HARDENED_TRY(otbn_load_app(kOtbnAppEcdh));
+
+  // Set mode so start() will jump into sideloaded keygen.
+  HARDENED_TRY(otbn_dmem_write(
+      kOtbnEcdhModeWords, &kOtbnEcdhModeKeypairFromSeed, kOtbnVarEcdhMode));
+
+  // Start the OTBN routine.
+  return otbn_execute();
+}
+
+status_t ecdh_p256_sideload_keypair_finalize(p256_point_t *public_key) {
+  // Spin here waiting for OTBN to complete.
+  HARDENED_TRY(otbn_busy_wait_for_done());
+
+  // Read the public key from OTBN dmem.
+  HARDENED_TRY(otbn_dmem_read(kP256CoordWords, kOtbnVarEcdhX, public_key->x));
+  HARDENED_TRY(otbn_dmem_read(kP256CoordWords, kOtbnVarEcdhY, public_key->y));
+
+  // Wipe DMEM.
+  HARDENED_TRY(otbn_dmem_sec_wipe());
```

Comment:
Should this function attempt to call `otbn_dmem_sec_wipe()` regardless of the outcome in `status_t`? If this is the case, then we may have to move most steps to a helper function. 

We should also update the function documentation to capture any potential cleanup tasks required by the caller in case of error.

Reply:
Wiping DMEM anyway in case of an error seems pretty reasonable. (The only way `otbn_dmem_read` can fail is if the read goes past the end of IMEM/DMEM, which really shouldn't happen except in the case of a fault attack, since the offset and length are both hardcoded here.)

However, I want to keep the codebase consistent and right now all the things that call OTBN do the wipe only if the reads succeed; I think that would be too much for this PR, so I'll make an issue to change this across all the ECC/RSA/software hash functions.

Reply:
https://github.com/lowRISC/opentitan/issues/20050

Reply:
Thanks!